File: app/src/main/java/com/arseniy/maps/data/CityAdapter.java
package com.arseniy.maps.data;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.arseniy.maps.R;

import java.util.ArrayList;

public class CityAdapter extends RecyclerView.Adapter<CityAdapter.CityViewHolder> {

    private ArrayList<String> cityNames;

    static class CityViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        final TextView textView;

        CityViewHolder(TextView v) {
            super(v);
            textView = v;
        }
    }

    public ArrayList<String> getData() {
        return cityNames;
    }

    public void setData(ArrayList<String> newArrayList) {
        cityNames = newArrayList;
        cityNames.sort(String::compareTo);
        notifyDataSetChanged();
    }

    public CityAdapter(ArrayList<String> cities) {
        cityNames = cities;
        cityNames.sort(String::compareTo);
    }


    @NonNull
    @Override
    public CityAdapter.CityViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // create a new view
        TextView v = (TextView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.city_text_view, parent, false);
        return new CityViewHolder(v);
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(@NonNull CityViewHolder holder, int position) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        holder.textView.setText(cityNames.get(position));

    }

    // Return the size of your dataset (invoked by the layout manager)
    @Override
    public int getItemCount() {
        return cityNames.size();
    }

}

File: app/src/main/java/com/arseniy/maps/data/CityAdapter.kt
package com.arseniy.maps.data

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.arseniy.maps.R

import java.util.ArrayList

class CityAdapter(private var cityNames: ArrayList<String>?) : RecyclerView.Adapter<CityAdapter.CityViewHolder>() {

    var data: ArrayList<String>?
        get() = cityNames
        set(newArrayList) {
            cityNames = newArrayList
            cityNames!!.sortWith(Comparator { obj, anotherString -> obj.compareTo(anotherString) })
            notifyDataSetChanged()
        }

    init {
        cityNames!!.sortWith(Comparator { obj, anotherString -> obj.compareTo(anotherString) })
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CityViewHolder {
        // create a new view
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.city_text_view, parent, false) as TextView
        return CityViewHolder(v)
    }

    // Replace the contents of a view (invoked by the layout manager)
    override fun onBindViewHolder(holder: CityViewHolder, position: Int) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        holder.textView.text = cityNames!![position]

    }

    // Return the size of your dataset (invoked by the layout manager)
    override fun getItemCount(): Int {
        return cityNames!!.size
    }

    class CityViewHolder(// each data item is just a string in this case
            val textView: TextView) : RecyclerView.ViewHolder(textView)

}